cpp/src/arrow/engine/CMakeLists.txt

```diff
@@ -46,14 +46,18 @@ add_arrow_lib(arrow_substrait
               SHARED_LINK_FLAGS
               ${ARROW_VERSION_SCRIPT_FLAGS} # Defined in cpp/arrow/CMakeLists.txt
               SHARED_LINK_LIBS
+              arrow_acero_shared
               arrow_dataset_shared
               substrait
               SHARED_INSTALL_INTERFACE_LIBS
+              ArrowAcero::arrow_acero_shared
```

Comment:
I think that we don't need this because `ArrowDataset::arrow_dataset_shared` has this dependency.